src/sds.c

```diff
@@ -266,7 +268,7 @@ sds _sdsMakeRoomFor(sds s, size_t addlen, int greedy) {
     if (type == SDS_TYPE_5) type = SDS_TYPE_8;
 
     hdrlen = sdsHdrSize(type);
-    assert(hdrlen + newlen + 1 > len);  /* Catch size_t overflow */
+    assert(hdrlen + newlen + 1 > newlen);  /* Catch size_t overflow */
```

Comment:
I don't think we should remove the old assertion.
Perhaps `assert(hdrlen + newlen + 1 > len && hdrlen + newlen + 1 > newlen);` would be more appropriate?

Reply:
> I don't think we should remove the old assertion.
> Perhaps `assert(hdrlen + newlen + 1 > len && hdrlen + newlen + 1 > newlen);` would be more appropriate?

Checking for `hdrlen + newlen + 1 > len` is not necessary here (Notice that at Line 253 we have `assert(newlen > len)`), since `assert(hdrlen + newlen + 1 > newlen)` can ensure the computation `hdrlen + newlen + 1` will not overflow. Also, we have add similar checks after each step of length computation, so the protection is correct.

Reply:
I have another idea.
We can add a temp variable to store the old `newlen`, which will reduce an assert.
your implementation is reasonable, and I'm not sure which is better.
ping @oranagra 

```
    origin_newlen = newlen = (len+addlen);
    ...
    assert(hdrlen + newlen + 1 > origin_newlen);  /* Catch size_t overflow */
```

Reply:
i like @sundb's suggestion (less assertions, possibly better performance and smaller assembly code).
so we have one assertion about appending `len` and `addlen`, and then another assertion after we add max of `1024*1024+1`.
let's name them `newlen` and `alloclen` (i think `origin` can be confusing)

Reply:
> i like @sundb's suggestion (less assertions, possibly better performance and smaller assembly code).
> so we have one assertion about appending `len` and `addlen`, and then another assertion after we add max of `1024*1024+1`.
> let's name them `newlen` and `alloclen` (i think `origin` can be confusing)



> i like @sundb's suggestion (less assertions, possibly better performance and smaller assembly code).
> so we have one assertion about appending `len` and `addlen`, and then another assertion after we add max of `1024*1024+1`.
> let's name them `newlen` and `alloclen` (i think `origin` can be confusing)

I have made an update to the commits. Please see if it is Ok.

Reply:
i meant the other way around, i.e. all the code below the `len+addlen` (except the assertion) to be changed to refer to `alloclen` rather than `newlen`.

but i see that's causing too many changes, and we can avoid that if we choose a new name for the other one.
so maybe rename `alloclen` in your code to `reqlen` (requested length / required length) it'll be good.

what i didn't like about the term `origin` is that it wasn't clear which one is the original one (arguably `len` is the original one).

Reply:
> i meant the other way around, i.e. all the code below the `len+addlen` (except the assertion) to be changed to refer to `alloclen` rather than `newlen`.
> 
> but i see that's causing too many changes, and we can avoid that if we choose a new name for the other one.
> so maybe rename `alloclen` in your code to `reqlen` (requested length / required length) it'll be good.
> 
> what i didn't like about the term `origin` is that it wasn't clear which one is the original one (arguably `len` is the original one).

Ok, I renamed `alloclen` to `reqlen`.